### PR TITLE
ignore

### PR DESCRIPTION
**Pull request by yozohu**

### CODE REVIEW

File: bellboy/actors/lcd.py [hunk 1]
```diff
@@ -1,8 +1,128 @@
...
+    
+    def displayLoop(self):
+        """running lcd thread. Displays text on screen"""
+
+        self.status = LCDResp.DISPLAY
+
+        while self.threadOn:
+            try:
+            self.log.info(
+                str.format("LCD DISPLAYS <<{}>>", )
```
Comment (yozohu) [inline bellboy/actors/lcd.py]:
```
while self.threadOn:
    lcd.message(self._display_string)
```
 
no need for try

File: bellboy/actors/lcd.py [hunk 2]
```diff
@@ -1,8 +1,128 @@
...
 class LiquidCrystalActor(GenericActor):
+    """
+    Class for the LCD Display.
+
+    Contains a polling thread which can be run or stopped on message
+    request.
+    """
+
+    def __init__(self):
+        super().__init__()
```
Comment (yozohu) [inline bellboy/actors/lcd.py:20]: add vriable for the string text content; i.e
``self._display_string = None``

File: bellboy/actors/lcd.py [hunk 3]
```diff
@@ -1,8 +1,128 @@
...
+            )
+            self.send(sender, Response.UNAUTHORIZED)
+            return
+
+        if message.type == LCDReq.SETUP:
+            self._setup_LCD(
+                _sdaPin=self._sdaPin,
+                sclPin=self._sclPin
+            )
+
```
Comment (yozohu) [inline bellboy/actors/lcd.py]: add clause for if type is LCDReq.DISPLAY

File: bellboy/actors/lcd.py [hunk 4]
```diff
@@ -1,8 +1,128 @@
...
+    def _clear(self):
+        self._sdaPin = 0
+        self._sclPin = 0
+
+        self._LCD.close()
+
+        self._buffer.clear()
+        self.log.debug("cleared LCD")
+        self.status = Response.READY
+
```
Comment (yozohu) [inline bellboy/actors/lcd.py:77]: add private message `` updateDisplay(text: str, duration: int)``

File: bellboy/actors/lead.py [hunk 5]
```diff
@@ -27,10 +29,26 @@ def startBellboyLead(self):
             UltrasonicActor, globalName="ultrasonic"
         )
 
+        self.log.info("Starting all dependent actors...")
+        self.lcd = self.createActor(
+            LiquidCrystal, globalName="lcd"
+        )
+        
         # request to setup sensor
         self.send(
```
Comment (yozohu) [inline bellboy/actors/lead.py]: send a seperate msg to the lcd, this send is sending to the ultrasonic_sensor

File: bellboy/actors/lead.py [hunk 6]
```diff
@@ -101,6 +120,18 @@ def receiveMsg_SensorEventMsg(self, message, sender):
             self.log.debug("received 3 events, turning off sensor.")
             self.send(self.ultrasonic_sensor, SensorReq.STOP)
 
+    def receiveMsg_LCDResp(self, sender):
+        self.log.info(
+            str.format("Received message from" , self.nameOf(sender))
+        )
+
+    def receiveMsg_LCDEventMsg(self, message, sender):
```
Comment (yozohu) [inline bellboy/actors/lead.py]: LCD prob wont generate events cus its an output device, this function isnt needed

File: bellboy/utils/messages.py [hunk 7]
```diff
@@ -96,3 +96,36 @@ def __init__(self, eventType, eventData):
...
+class LCDMsg:
+    def __init__(self, msgType, LCDNumber=None):
+        self.msgType = msgType
+        self.LCDNumber = LCDNumber
+
+    def __str__(self):
+        return self.msgType.name
+
+
+class LCDEvent(Enum):
```
Comment (yozohu) [inline bellboy/utils/messages.py]: events stuff not needed

File: bellboy/utils/messages.py [hunk 8]
```diff
@@ -96,3 +96,36 @@ def __init__(self, eventType, eventData):
...
+
+class LCDReq(Enum):
+    SETUP, DISPLAY= range(2)
+
+
+class LCDResp(Enum):
+    SET, DISPLAYING= range(2)
+
+
+class LCDMsg:
```
Comment (yozohu) [inline bellboy/utils/messages.py]: whats LCDnnumber

File: bellboy/utils/messages.py [hunk 9]
```diff
@@ -96,3 +96,36 @@ def __init__(self, eventType, eventData):
...
+    def __init__(self, msgType, LCDNumber=None):
```
Comment (yozohu) [inline bellboy/utils/messages.py]: add parameters ``displayText (string)`` and ``displayDuration (int/long)`` for messages that will need those

**Review by yozohu — COMMENTED** [review]
comments!

File: bellboy/actors/lcd.py (hunk 4 above)
Comment (seinizumita) [reply bellboy/actors/lcd.py:77]: where does this go?

File: bellboy/actors/lcd.py (hunk 1 above)
Comment (RyanFleck) [reply bellboy/actors/lcd.py]: That's true, the alignment here is kind of wonky too.

File: bellboy/actors/lcd.py (hunk 4 above)
Comment (RyanFleck) [reply bellboy/actors/lcd.py:77]: @yozohu  doesn't Sein already have something like that at `System/bellboy/utils/messages.py ` line 125?

Comment (yozohu) [reply bellboy/actors/lcd.py:77]: ### not emssage i meant function sry

**Pull request closed without merging**